### PR TITLE
feat/e2e-test-pipeline

### PR DESCRIPTION
**Pull request by Fovty**
### Problem to solve

Creating a fully functional e2e pipeline

### Proposal

Clone sparrow e2e pipeline and adapt it accordingly
### Who can address the issue

- GitHub actions knowledge
- Kubernetes, Helm
### Other links/references

- e2e pipeline of sparrow: https://github.com/caas-team/sparrow/blob/main/.github/workflows/end2end.yml#L44

### CODE REVIEW

**Review by eumel8 — APPROVED** [review]
LGTM

File: chart/values.yaml
Comment (puffitos) [inline chart/values.yaml:6]: Is an image with the chart.AppVersion even created when the github actions run? We should build the image as well, when we create a new tag in the repository.

File: .github/workflows/docker-build.yml
Comment (puffitos) [inline .github/workflows/docker-build.yml:23]:
```
            echo "::set-output name=tag2::commit-$SHORT_SHA"
```

can be executed once, when we're pushing to the non-main branch. The latest tag is only needed when we do a tagged release actually. We should offer a `latest` tag, just because we've pushed something to main. We shouldn't even build an image when we merge to main, to be honest. The main branch should just depict the current state of the developement. It needn't offer a stable artifact (a docker image and a helm chart); that's what tagged commits are for.

**Review by puffitos — APPROVED** [review]
LGTM. Please open an issue regarding my reviews. 

We shouldn't build a latest image in the main branch, as the main branch doesn't always need to be fully functional, and directly tagging something with `latest` means that it will get pulled automatically by various runtimes. The `latest` image (as well as a properly versioned image) should be produced when git tags are created.

File: .github/workflows/docker-build.yml
Comment (Fovty) [reply .github/workflows/docker-build.yml:23]: Yeah, makes totally sense. Created #19

File: chart/values.yaml
Comment (Fovty) [reply chart/values.yaml:6]: Currently, the CI/CD process does not automatically create a Docker image tagged specifically with the chart.AppVersion. Mentioned it as well in #19

**Comment by puffitos** [comment]
@jonathan-mayer @JTaeuber waiting on your approval here, when you got some time. 

@Fovty Thanks for creating the issue 👍

**Review by JTaeuber — APPROVED** [review]
LGTM

**Pull request merged**
